packages/graphql/src/translate/queryAST/ast/QueryASTContext.ts

```diff
@@ -37,4 +37,12 @@ export class QueryASTContext {
         this.relationship = relationship;
         this.source = source;
     }
+
+    public push({ relationship, target }: { relationship: Cypher.Relationship; target: Cypher.Node }): QueryASTContext {
```

Comment:
I think here push here it's a little bit ambiguous as it seems that we push a new Context in some stack when in reality we just return a new QueryASTContext

Reply:
:thinking: 

The idea was kind of ("push a new relationship into the stack"), but maybe it is not needed and we can find a better way